azurerm/resource_arm_automation_schedule.go

```diff
@@ -282,3 +364,90 @@ func resourceArmAutomationScheduleDelete(d *schema.ResourceData, meta interface{
 
 	return nil
 }
+
+func expandArmAutomationScheduleAdvanced(v []interface{}) (*automation.AdvancedSchedule, error) {
+
+	// Get the one and only advance schedule configuration
+	advancedSchedule := v[0].(map[string]interface{})
+	weekDays := advancedSchedule["week_days"].(*schema.Set).List()
+	monthDays := advancedSchedule["month_days"].(*schema.Set).List()
+	monthlyOccurrences := advancedSchedule["monthly_occurrence"].([]interface{})
+
+	expandedAdvancedSchedule := automation.AdvancedSchedule{}
+
+	if len(weekDays) > 0 {
+		expandedWeekDays := make([]string, len(weekDays))
+		for i := range weekDays {
+			expandedWeekDays[i] = weekDays[i].(string)
+		}
+		expandedAdvancedSchedule.WeekDays = &expandedWeekDays
+	}
+
+	if len(monthDays) > 0 {
```

Comment:
can we remove this if statement - since it means it's not possible to remove these values once they've been set.

Reply:
Same as with `weekDays`.